.github/workflows/create-release-tag.yaml

```
@@ -0,0 +1,60 @@
+name: tag-generation
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+    paths:
+      - 'CHANGELOG.md'
+
+jobs:
+  create-tag:
+    if: github.event.pull_request.merged == true && contains(join(github.event.pull_request.labels.*.name, ','), 'release') && startsWith(github.event.pull_request.title, 'Update changelog to prep for release')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Check if PR author is a maintainer
+        run: |
+          pr_author="${{ github.event.pull_request.user.login }}"
+          permission=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/${{ github.repository }}/collaborators/$pr_author/permission" | jq -r .permission)
+
+          if [[ "$permission" == "admin" || "$permission" == "write" ]]; then
```

```suggestion
          if [[ "$permission" == "admin" || "$permission" == "maintain" || "$permission" == "write" ]]; then
```

Maintain > write and should be included if write is included

---

Q: Why does it matter if the PR author is a maintainer? Maintainers are the ones who will be ultimately reviewing and merging the PR.

Community members should still be able to help by e.g. cleaning up release notes and opening up the PRs.

---

@poovamraj ^

---

Yep, that makes sense 👍

---

@rhamzeh I think we should keep releases contained to maintainers. I understand your point. But we should cross that point when we get there.

Its better to have tighter restrictions and opening up later based on the need